scipy/signal/_peak_finding_utils.pyx

```diff
@@ -74,3 +74,70 @@ def _argmaxima1d(np.float64_t[:] x not None):
 
     maxima.resize(m, refcheck=False)  # Keep only valid part of array memory.
     return maxima
+
+
+@cython.wraparound(False)
+@cython.boundscheck(False)
+def _select_by_peak_distance(np.intp_t[:] peaks not None,
+                             np.float64_t[:] priority not None,
+                             dmin):
+    """
+    Evaluate which peaks fulfill the distance condition.
+
+    Parameters
+    ----------
+    peaks : ndarray
+        Indices of peaks in `vector`.
+    priority : ndarray
+        An array matching `peaks` used to determine priority of each peak. A peak
+        with a higher priority value is kept over one with a lower one.
+    dmin : number
```

Comment:
`number` is perhaps a slightly vague "type" -- is `dmin` supposed to be an integer or can it be a float as well?

Reply:
My impression from the code below is "natural number" -- so positive integers? Does that mean it could actually be typed as an unsigned integer?

Reply:
See my answer on you other comments.